usl_pipeline/tests/chunkers/test_elevation_chunkers.py

```diff
@@ -0,0 +1,165 @@
+import tempfile
+
+import numpy
+from numpy import testing
+from osgeo import gdal
+import rasterio
+
+from usl_pipeline.readers import elevation_readers
+from usl_pipeline.chunkers import elevation_chunkers
+from usl_pipeline.shared import geo_data
+from usl_pipeline.shared import suppliers
+
+
+def prepare_test_geotiff_elevation_memory_file(file_path: str):
+    tiff_array = numpy.array(
+        [
+            [0.0, 1.0, 2.0],
+            [3.0, 4.0, 5.0],
+            [6.0, 7.0, 8.0],
+        ]
+    )
+    with rasterio.open(
+        file_path,
+        "w",
+        driver="GTiff",
+        dtype=rasterio.float32,
+        nodata=0.0,
+        width=3,
+        height=3,
+        count=1,
+        crs=rasterio.CRS.from_epsg(32618),
+        transform=rasterio.Affine(2.0, 0.0, 10.0, 0.0, -2.0, 20.0),
+    ) as raster:
+        raster.write(tiff_array.astype(rasterio.float32), 1)
+
+
+class MemoryChunkPathGenerator(suppliers.ChunkFilePathGenerator):
```

Comment:
What is the `MemoryChunkPathGenerator` callers would use outside of tests?
I'm concerned the tests will be testing the implementation of `MemoryChunkPathGenerator` rather than the implementation callers will actually be using.

I'd expect the non-test implementation would either expect a `file: typing.BinaryIO`  or a `file_path: str | pathlib.Path`. I'd rather work with that implementation itself in the tests to make sure we're testing the implementation real callers will actually use, even if that means having to work more with temporary files than memory files.